core/client/src/main/java/alluxio/client/lineage/AbstractLineageClient.java

```diff
@@ -63,12 +65,19 @@ public long createLineage(List<AlluxioURI> inputFiles, List<AlluxioURI> outputFi
     // TODO(yupeng): relax this to support other type of jobs
     Preconditions.checkState(job instanceof CommandLineJob,
         PreconditionMessage.COMMAND_LINE_LINEAGE_ONLY);
-    LineageMasterClient masterClient = mContext.acquireMasterClient();
+    LineageMasterClient masterClient = null;
     try {
+      masterClient = mContext.acquireMasterClient();
       long lineageId = masterClient.createLineage(stripURIList(inputFiles),
           stripURIList(outputFiles), (CommandLineJob) job);
       LOG.info("Created lineage {}", lineageId);
       return lineageId;
+    } catch (NotFoundException e) {
+      throw new FileDoesNotExistException(e.getMessage());
+    } catch (UnavailableException e) {
+      throw new IOException(e.getMessage());
+    } catch (AlluxioStatusException e) {
+      throw e.toAlluxioException();
```

Comment:
sorry, missed this comment

For users `IOException` usually means "Exception that might be worth retrying", which is also the meaning of `UnavailableException`. Therefore, we convert `UnavailableException` to `IOException` to preserve this. Other types of status exception are generally not worth retrying, so we convert them to `AlluxioException`, which previously meant "something that went wrong in Alluxio that isn't IOException"